### PR TITLE
Fix FlowMatchEulerDiscreteScheduler double-shift in set_timesteps

### LINKED ISSUE #13243: [BUG] FlowMatchEulerDiscreteScheduler.__init__ computes sigma_min/sigma_max after shift, causing duplicate shift in set_timesteps
### Describe the bug

There is an initialization order issue in the FlowMatchEulerDiscreteScheduler class that causes set_timesteps to apply timestep shifting twice, resulting in different sigma values for the same timestep settings.
https://github.com/huggingface/diffusers/blob/07a63e197e10860a470576cf4f610381b31a4dd7/src/diffusers/schedulers/scheduling_flow_match_euler_discrete.py#L132-L143
Then in the set_timesteps method:
https://github.com/huggingface/diffusers/blob/07a63e197e10860a470576cf4f610381b31a4dd7/src/diffusers/schedulers/scheduling_flow_match_euler_discrete.py#L333-L350

sigma_min will be saved after shift and used to generate a new list of sigma, then there is another shift happening :).

### Reproduction

scheduler = FlowMatchEulerDiscreteScheduler.from_pretrained(
            os.path.join(model_path, "scheduler"),
        )
# default scheduler.num_train_timesteps=1000,shift=3.0
print(scheduler.sigmas)
scheduler.set_timesteps(1000)
print(scheduler.sigmas)

### Logs

```shell

```

### System Info

Environment
diffusers version
Python 3.13
Affected file: diffusers/schedulers/scheduling_flow_match_euler_discrete.py

### Who can help?

_No response_

### PR DESCRIPTION
**Pull request by Ricardo-M-L**
## What does this PR do?

Fixes #13243 — `FlowMatchEulerDiscreteScheduler.set_timesteps` applies the static timestep shift a second time when `use_dynamic_shifting=False`, so calling `set_timesteps(num_train_timesteps)` on a scheduler produces a different sigma schedule than the one `__init__` cached.

### Root cause

`__init__` records `sigma_min`/`sigma_max` from the **post-shift** sigma range:

```python
sigmas = timesteps / num_train_timesteps
if not use_dynamic_shifting:
    sigmas = shift * sigmas / (1 + (shift - 1) * sigmas)  # ← shift applied
...
self.sigma_min = self.sigmas[-1].item()  # ← post-shift value
self.sigma_max = self.sigmas[0].item()   # ← post-shift value
```

`set_timesteps` then rebuilds sigmas via `linspace(_sigma_to_t(sigma_max), _sigma_to_t(sigma_min), N)` — already in shifted space — and applies the same shift again:

```python
sigmas = self.shift * sigmas / (1 + (self.shift - 1) * sigmas)
```

### Reproduction

```python
from diffusers import FlowMatchEulerDiscreteScheduler
import torch

s = FlowMatchEulerDiscreteScheduler(num_train_timesteps=1000, shift=3.0, use_dynamic_shifting=False)
init_sigmas = s.sigmas.clone()
s.set_timesteps(num_inference_steps=1000)

print(init_sigmas[-3:])       # tensor([0.0089, 0.0060, 0.0030])
print(s.sigmas[-4:-1])        # tensor([0.0148, 0.0119, 0.0089])  ← different
torch.allclose(init_sigmas, s.sigmas[:-1])  # False
```

The same args produce two different schedules.

### Fix

Record the pre-shift sigmas in `__init__` and derive `sigma_min`/`sigma_max` from them. `set_timesteps` then regenerates the linear sigma range correctly and the static shift is applied exactly once. `self.sigmas` is unchanged (still post-shift); only the cached endpoints change.

### Test

Added `tests/schedulers/test_scheduler_flow_match_euler_discrete.py`:

- `test_set_timesteps_matches_init_with_static_shift` — fails on `main`, passes with the fix.
- `test_dynamic_shifting_is_unaffected` — pins that the `use_dynamic_shifting=True` path is untouched (no static shift in `__init__`, so this code path never had the bug).

## Before submitting
- [x] This PR fixes a bug — links #13243 with a clear repro.
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you write any new necessary tests? — Yes, regression test added.

## Who can review?

@yiyixuxu @DN6

### CODE REVIEW

**Comment by Ricardo-M-L** [comment]
Hi team, just following up on this PR. Happy to make any changes or address feedback if needed. Thanks for considering it!

**Comment by Ricardo-M-L** [comment]
Hi! Just a friendly ping on this PR. It's been open for a while — would appreciate a review when you get a chance. Thanks for your time!

**Comment by Ricardo-M-L** [comment]
Hi folks! Friendly ping on this FlowMatch scheduler fix. The double-shift in set_timesteps was causing incorrect timestep spacing. It is a small, targeted fix with no API changes. Would appreciate a review whenever you have a moment. Thanks!

**Comment by Ricardo-M-L** [comment]
Friendly ping - this PR has been open for a while. Would appreciate a review when you get a chance. Thanks!

**Comment by yiyixuxu** [comment]
hi @Ricardo-M-L

I'm closing this per the AI-assisted contribution guidelines: https://huggingface.co/docs/diffusers/main/en/conceptual/contribution#ai-assisted-and-agentic-contributions

We're seeing an increasing volume of contributions and review capacity is currently our bottleneck, so the guidelines ask for maintainer acknowledgment on an issue before a PR is opened — that's what lets us prioritize the bugs users are actually hitting.

Normally we'd leave this as a suggestion and keep working with a contributor on the PR, but the volume here has made that impractical — these have come in faster than we can review them, so I'm closing all of them that you've opened rather than reviewing them individually.
